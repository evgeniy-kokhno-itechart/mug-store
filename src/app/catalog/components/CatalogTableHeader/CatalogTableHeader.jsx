import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { SearchBox } from '../../../shared';
import SortBox from '../SortBox/SortBox';
import './CatalogTableHeader.scss';

const CatalogTableHeader = ({
  currentUserRoles, searchQuery, sortColumnKey, sortOptions, handleSearch, handleSort,
}) => (
  <div className="catalog__header">
    <div className="search-box">
      <SearchBox value={searchQuery} onSearchSubmit={handleSearch} key={searchQuery} />
    </div>
    {currentUserRoles.includes('admin') && (
      <Link to="/edit/products/new" className="button button-solid add-product-btn">
        Add New Product
      </Link>
    )}
    <div className="sort-box">
      <SortBox sortColumnKey={sortColumnKey} sortOptions={sortOptions} onSortChange={handleSort} />
    </div>
  </div>
);

CatalogTableHeader.propTypes = {
  currentUserRoles: PropTypes.arrayOf(PropTypes.string),
  searchQuery: PropTypes.string,
  sortColumnKey: PropTypes.string.isRequired,
  sortOptions: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  handleSearch: PropTypes.func.isRequired,
  handleSort: PropTypes.func.isRequired,
};

CatalogTableHeader.defaultProps = {
  currentUserRoles: [],
  searchQuery: '',
};

export default CatalogTableHeader;
