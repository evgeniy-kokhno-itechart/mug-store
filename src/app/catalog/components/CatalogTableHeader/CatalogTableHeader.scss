@import "Mixins";

.catalog {
    &__header {
        display: flex;
        justify-content: space-between;
        flex-wrap: wrap;
        align-items: center;
        margin: 0 1rem 1rem;

        @include respond(tab-portrait) {
            flex-direction: column;
            width: 70%;
            margin: 0 auto;
        }
    }

    &-search-box {
        @include respond(tab-portrait) {
            width: 100%;
            margin: 0.5rem 0;
        }
    }

    &-add-product-btn {
        white-space: nowrap;
        min-width: 10rem;
        padding-right: 3rem;
        padding-left: 3rem;

        @include respond(tab-portrait) {
            width: 100%;
            margin: 0.5rem 0;
        }
    }

    &-sort-box {
        margin: 0;

        @include respond(tab-portrait) {
            width: 70%;
            margin: 0.5rem auto;
            text-align: center;
        }
    }
}
