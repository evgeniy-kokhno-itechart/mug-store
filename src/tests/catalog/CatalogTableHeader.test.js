/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CatalogTableHeader from '../../app/catalog/components/CatalogTableHeader';

configure({ adapter: new Adapter() });

describe('<CatalogTableHeader />', () => {
  let CatalogTableHeaderWrapper;

  beforeEach(() => {
    const handleSearch = jest.fn();
    const handleSort = jest.fn();
    CatalogTableHeaderWrapper = shallow(
      <CatalogTableHeader
        sortColumnKey="title"
        sortOptions={[{ id: 'title_desc', name: 'Title Z-A' }]}
        handleSearch={handleSearch}
        handleSort={handleSort}
      />,
    );
  });

  it('check render properly', () => {
    expect(CatalogTableHeaderWrapper).toMatchSnapshot();
  });

  it('render AddNewProduct button with currentUserRole admin provided', () => {
    CatalogTableHeaderWrapper.setProps({ currentUserRoles: ['admin'] });
    expect(CatalogTableHeaderWrapper).toMatchSnapshot();
  });

  it('check render with searchQuery provided', () => {
    CatalogTableHeaderWrapper.setProps({ searchQuery: 'test query' });
    expect(CatalogTableHeaderWrapper).toMatchSnapshot();
  });
});
