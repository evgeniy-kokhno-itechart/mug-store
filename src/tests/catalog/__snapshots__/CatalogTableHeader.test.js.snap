// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogTableHeader /> check render properly 1`] = `
<div
  className="row justify-content-between mx-1 mx-sm-2"
>
  <div
    className="col-12 col-md-5 col-lg-4 col-xl-3"
  >
    <SearchBox
      key=""
      onSearchSubmit={[MockFunction]}
      value=""
    />
  </div>
  <div
    className="col-12 col-md col-lg-3 col-xl-2"
  >
    <SortBox
      onSortChange={[MockFunction]}
      sortColumnKey="title"
      sortOptions={
        Array [
          Object {
            "id": "title_desc",
            "name": "Title Z-A",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`<CatalogTableHeader /> check render with searchQuery provided 1`] = `
<div
  className="row justify-content-between mx-1 mx-sm-2"
>
  <div
    className="col-12 col-md-5 col-lg-4 col-xl-3"
  >
    <SearchBox
      key="test query"
      onSearchSubmit={[MockFunction]}
      value="test query"
    />
  </div>
  <div
    className="col-12 col-md col-lg-3 col-xl-2"
  >
    <SortBox
      onSortChange={[MockFunction]}
      sortColumnKey="title"
      sortOptions={
        Array [
          Object {
            "id": "title_desc",
            "name": "Title Z-A",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`<CatalogTableHeader /> render AddNewProduct button with currentUserRole admin provided 1`] = `
<div
  className="row justify-content-between mx-1 mx-sm-2"
>
  <div
    className="col-12 col-md-5 col-lg-4 col-xl-3"
  >
    <SearchBox
      key=""
      onSearchSubmit={[MockFunction]}
      value=""
    />
  </div>
  <div
    className="col-12 col-md-4 col-lg-3 col-xl-3"
  >
    <Link
      className="btn btn-secondary mt-1 mt-md-0 w-100 px-md-2"
      to="/edit/products/new"
    >
      Add New Product
    </Link>
  </div>
  <div
    className="col-12 col-md col-lg-3 col-xl-2"
  >
    <SortBox
      onSortChange={[MockFunction]}
      sortColumnKey="title"
      sortOptions={
        Array [
          Object {
            "id": "title_desc",
            "name": "Title Z-A",
          },
        ]
      }
    />
  </div>
</div>
`;
